src/vaadin-message-input.js

```diff
@@ -65,6 +79,29 @@ class MessageInputElement extends ElementMixin(ThemableMixin(PolymerElement)) {
     // Set initial height to one row
     textarea.setAttribute('rows', 1);
     textarea.style.minHeight = '0';
+
+    // Add enter handling for text area.
+    textarea.addEventListener('keydown', (e) => {
+      if (e.key === 'Enter' && !e.shiftKey) {
+        e.preventDefault();
+        e.stopPropagation();
+        this.__submit();
+      }
+    });
+  }
+
+  /**
+   * Submits the current value as an custom event named 'submit'.
+   * It also clears the text input and refocuses it for sending another message.
+   * In UI, can be triggered by pressing the submit button or pressing enter key when field is focused.
+   * It does not submit anything if text is empty.
+   */
+  __submit() {
+    if (this.value !== '') {
+      this.dispatchEvent(new CustomEvent('submit', { detail: { text: this.value } }));
```

Comment:
I think `value` makes sense here as well:
```suggestion
      this.dispatchEvent(new CustomEvent('submit', { detail: { value: this.value } }));
```